src/web/common.rkt

```diff
@@ -8,6 +8,18 @@
   (match-define (list _ args expr) prog)
   (list 'FPCore args expr))
 
+(define (fpcore-add-props core props)
+  (match-define (list 'FPCore args expr) core)
+  `(FPCore ,args ,@props ,expr))
+
+(define (fpcore->string core)
+  (match-define (list 'FPCore args props ... expr) core)
+  (define props*
+    (for/list ([val (apply dict-set* '() props)])
+      (format "~a ~a" (car val) (cdr val))))
+  (with-output-to-string
+    (λ () (pretty-display `(,(format "FPCore ~a" args) ,@props* ,expr)))))
```

Comment:
This is both very clever and horrible. At the least there should be a comment. Also, you want want `pretty-format` with `#:mode 'display` instead of `with-output-to-string`.